drivers/media/platform/meson/meson_drv.c

```diff
@@ -1291,9 +1291,26 @@ static int queue_init(void *priv, struct vb2_queue *src_vq, struct vb2_queue *ds
 	return vb2_queue_init(dst_vq);
 }
 
-static bool image_buffers_ready(struct vdec_ctx *ctx) {
+static bool image_buffers_ready(struct vdec_ctx *ctx)
+{
+	struct buffer_size_info buf_info;
+	struct vb2_queue *vq;
+	bool dst_size_ok = true;
+	int i;
+
+	vq = v4l2_m2m_get_vq(ctx->m2m_ctx, V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE);
+	get_buffer_size_info(ctx->fmt, ctx->frame_width,
+			     ctx->frame_height, &buf_info);
+
+	for (i = 0; i < buf_info.num_planes; i++)
+		if (buf_info.buffer_size[i] != vq->plane_sizes[i]) {
+			dst_size_ok = false;
```

Comment:
I added that check to not change the code path when (if ever) decoding H264 instead of VP9/WEBM. For the H264 we have never seen this issue and even though the fix is formally valid also in that case I thought it was better to skip the H264 case.